### ISSUE TITLE
TrayIcon content does not react to DynamicResource changes

### ISSUE THREAD

**Issue by NikasZalias** [hardcodet/wpf-notifyicon#19]
How can I make my TrayIcon react to Theme changes? When I change my app theme from light to dark or vice versa, tray icon stays the same until I restart app. Why is it like that?

**Comment by NikasZalias** [comment]
@punker76

**Comment by angelwzr** [comment]
I'm interested in finding out what causes this behavior too.
@NikasZalias were you able to fix this after all?

**Comment by sirdoombox** [comment]
I'm also having similar issues using MahApps styles for the ContextMenu and MenuItem, for some reason only the foreground hover brush is changed when you change themes, but the foreground/background is not.

**Comment by Symbai** [comment]
Same

**Comment by CoelWu** [comment]
Same here, I'm settings MenuItem's header to a dynamicResource, it just does not change when the dynamicResource changes.

**Comment by CoelWu** [comment]
@punker76

**Comment by Lakritzator** [comment]
Maybe someone can make a minimalistic example project for this?

**Comment by Erapchu** [comment]
As a solution: subscribe to TrayContextMenuOpen event of TaskbarIcon, and call UpdateDefaultStyle for context menu that required to be updated. Or call UpdateDefaultStyle when theme changed.

**Comment by Erapchu** [comment]
Popup control that host context menu or other controls can't correctly get dynamic resources changes from Application merged resource I guess. Popup on the same visual level where Application exists.

**Comment by DarkPro1337** [comment]
> As a solution: subscribe to TrayContextMenuOpen event of TaskbarIcon, and call UpdateDefaultStyle for context menu that required to be updated. Or call UpdateDefaultStyle when theme changed.

Working as intended now, much thanks!

**Comment by Lakritzator** [comment]
I need to look into this again, I am also faced with the issue, so maybe I can add something to support a style update.

Btw. example link mentioned above moved to here: https://github.com/Erapchu/SinglePass.WPF/blob/master/SinglePass.WPF/Views/TrayIcon.xaml.cs

**Comment by Erapchu** [comment]
I renamed repo, thanks @Lakritzator 
Want to leave this solution here, if anyone need it

```
private void TaskbarIcon_TrayContextMenuOpen(object sender, RoutedEventArgs e)
{
    // To apply DynamicResource changes
    TrayIconContextMenu?.UpdateDefaultStyle();
}
```

**Comment by punker76** [comment]
@NikasZalias This should be fixed now with #117 by using @Erapchu solution
